CDCgov/prime-reportstream
# 7561 Updates to HL7 to FHIR mappings for MSH segment

This PR updates the Bundle and Message Header resources from the HL7 v2 MSH segment using the latest HL7 v2 to FHIR mapping documents from HL7. Updates to the Provenance resource will come as another PR for this issue.

A copy of our updated MSH spreadsheet is here (master is in SharePoint):
[HL7 to FHIR - MSH Segment.xlsx](https://github.com/CDCgov/prime-reportstream/files/10296414/HL7.to.FHIR.-.MSH.Segment.xlsx)
 
Test Steps:
1. Build this branch and run the unit and integration tests.  Verify they all pass.  Alternatively you can verify the Github Actions build passes. Note the HL7 to FHIR to HL7 integration test data was not changed, so the result of our HL7 to FHIR to HL7 conversion remains the same = no data changes.
2. Optional: Manually change some IDs in MSH 2 through 5 to be UUID, DNS, CLIA, URI and verify the resulting FHIR is correct.  Refer to the mapping sheets to know which fields to change as not all fields are meant to support all types.

## Changes
- Added detection of DNS, URI and UUID IDs
- Moved out custom FHIR Path functions to its own file as the code was getting big
- Added missing tests to some custom FHIR Path functions
- Updated HL7 v2 to FHIR mappings
- Updated FHIR to HL7 v2 mappings
- Updated FHIR files in integration tests

## Checklist

### Testing

### Process

## Linked Issues
- #7561

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
